### ISSUE TITLE
VirtualBox plugin notworking

### ISSUE THREAD

**Issue by comio** [OpenMediaVault-Plugin-Developers/openmediavault-virtualbox#11]
The last openmediavault release breaks the Virtualbox Plugin:

```

ii  openmediavault                     3.0.24                               all          Open network attached storage solution
ii  openmediavault-autoshutdown        3.1.1                                all          OpenMediaVault AutoShutdown Plugin
rc  openmediavault-flashmemory         3.2                                  all          folder2ram plugin for OpenMediaVault
ii  openmediavault-keyring             0.4                                  all          GnuPG archive keys of the OpenMediaVault archive
ii  openmediavault-omvextrasorg        3.3.3                                all          OMV-Extras.org Package Repositories for OpenMediaVault
ii  openmediavault-virtualbox          3.0.4                                all          VirtualBox plugin for OpenMediaVault.
rc  openmediavault-wakealarm           3.1                                  all          wakealarm plugin for OpenMediaVault.
ii  openmediavault-omvextrasorg        3.3.3                                all          OMV-Extras.org Package Repositories for OpenMediaVault
ii  openmediavault-virtualbox          3.0.4                                all          VirtualBox plugin for OpenMediaVault.
ii  phpvirtualbox                      5.0-5-1                              all          Web interface for VirtualBox.
ii  virtualbox                         5.0.20-dfsg-2~bpo8+1                 amd64        x86 virtualization solution - base binaries
ii  virtualbox-dkms                    5.0.20-dfsg-2~bpo8+1                 all          x86 virtualization solution - kernel module sources for dkms
ii  virtualbox-ext-pack-installer      5.0.16                               all          VirtualBox extension pack installer.

```

The omv-rpc call inside config.php fails causing the issue (also blocking phpvirtualbox):
Calling "omv-rpc" from command line:
```

/usr/sbin/omv-rpc  VirtualBox getSettings
PHP Notice:  Use of undefined constant OMV_ROLE_ADMINISTRATOR - assumed 'OMV_ROLE_ADMINISTRATOR' in /usr/sbin/omv-rpc on line 85
PHP Notice:  Use of undefined constant OMV_RPC_MODE_REMOTE - assumed 'OMV_RPC_MODE_REMOTE' in /usr/sbin/omv-rpc on line 106
PHP Notice:  Use of undefined constant OMV_RPC_MODE_LOCAL - assumed 'OMV_RPC_MODE_LOCAL' in /usr/share/php/openmediavault/rpc/rpc.inc on line 68
PHP Notice:  Use of undefined constant OMV_RPC_MODE_REMOTE - assumed 'OMV_RPC_MODE_REMOTE' in /usr/share/php/openmediavault/rpc/rpc.inc on line 93
{"response":null,"error":{"code":0,"message":"RPC service 'VirtualBox' not found.","trace":"exception 'OMV\\Rpc\\Exception' with message 'RPC service 'VirtualBox' not found.' in \/usr\/share\/php\/openmediavault\/rpc\/rpc.inc:94\nStack trace:\n#0 \/usr\/sbin\/omv-rpc(106): OMV\\Rpc\\Rpc::call('VirtualBox', 'getSettings', NULL, Array, 'OMV_RPC_MODE_RE...')\n#1 {main}"}}

```

This the error from omv gui:

```


Failed to get OMV VirtualBox settings.
Errore #0:
exception 'Exception' with message 'Failed to get OMV VirtualBox settings.' in /usr/share/phpvirtualbox/config.php:104
Stack trace:
#0 /usr/share/phpvirtualbox/config.php(81): phpVBoxConfig->getSettings()
#1 /usr/share/phpvirtualbox/endpoints/lib/config.php(156): phpVBoxConfig->__construct()
#2 /usr/share/phpvirtualbox/endpoints/lib/vboxconnector.php(128): phpVBoxConfigClass->__construct()
#3 /usr/share/openmediavault/engined/rpc/virtualbox.inc(134): vboxconnector->__construct()
#4 [internal function]: OMV\Engined\Rpc\VirtualBox->getMachines(Array, Array)
#5 /usr/share/php/openmediavault/rpc/serviceabstract.inc(124): call_user_func_array(Array, Array)
#6 /usr/share/php/openmediavault/rpc/rpc.inc(81): OMV\Rpc\ServiceAbstract->callMethod('getMachines', Array, Array)
#7 /usr/sbin/omv-engined(524): OMV\Rpc\Rpc::call('VirtualBox', 'getMachines', Array, Array, 1)
#8 {main}

```

Thanks for your effort.

luigi

**Comment by comio** [comment]
http://forums.openmediavault.org/index.php/Thread/14209-Failed-to-get-OMV-VirtualBox-settings/?postID=117288#post117288

(fix must be placed into omv-rpc)

**Comment by ryecoaaron** [comment]
Fixed in 3.0.5.
